src/main/java/com/boozallen/aissemble/UploadDependenciesMojo.java

```diff
@@ -0,0 +1,62 @@
+package com.boozallen.aissemble;
+
+import java.io.File;
+import java.nio.file.Paths;
+
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.project.MavenProject;
+
+import io.kokuwa.maven.helm.UploadMojo;
+import io.kokuwa.maven.helm.pojo.HelmRepository;
+import io.kokuwa.maven.helm.pojo.RepoType;
+
+@Mojo(name = "upload-dependencies")
+public class UploadDependenciesMojo extends UploadMojo {
+
+	@Parameter(property = "url", required = true)
+	private String url;
+
+	@Parameter(property = "id", required = true)
+	private String id;
+
+	@Parameter(property = "type", required = true)
+	private RepoType type;
+
+	@Parameter(defaultValue = "${project}", required = true, readonly = true)
+	private MavenProject project;
+
+	@Override
+	public void execute() throws MojoExecutionException {
+		String projectPath = project.getBasedir().getAbsolutePath();
+		if (!isHelmChart(projectPath)) {
+			return;
+		}
+		uploadRepoStable = createHelmRepository(id, url, type);
+		uploadRepoSnapshot = createHelmRepository(id, url, type);
+		outputDirectory = new File(Paths.get(projectPath, "charts").toString());
+		if (!outputDirectory.exists()) {
```

Comment:
S:  The charts directory may not exist or may be out-of-date if the `DependencyUpdateMojo` hasn't executed.